### ISSUE TITLE
vagrant + synchronize don't work.

### ISSUE THREAD

**Issue by bitozoid** [ansible/ansible#11419]
#5240 is happening to me. I patched like this to make it work:

``` patch
--- /home/bitozoid/synchronize.py   2015-06-29 15:42:05.831924630 +0100
+++ /usr/lib64/python2.7/site-packages/ansible/runner/action_plugins/synchronize.py 2015-06-29 15:37:36.752538804 +0100
@@ -145,7 +145,7 @@

         # COMPARE DELEGATE, HOST AND TRANSPORT
         process_args = False
-        if not dest_host is src_host and self.original_transport != 'local':
+        if ( ( not dest_host is src_host ) or dest_port != 22 ) and self.original_transport != 'local':
             # interpret and inject remote host info into src or dest
             process_args = True
```

```
$ vagrant --version
Vagrant 1.7.2
$ ansible --version
ansible 1.9.0.1
  configured module search path = None
```

**Comment by bcoca** [comment]
we might need a bit more complex a check, dest_port != ansible_ssh_port?

**Comment by nomaster** [comment]
On Ansible 2.0:

``` python
if ( ( not dest_host is src_host ) or dest_port != 22 ) and remote_transport:
```

**Comment by simono** [comment]
I'm seeing this error with Vagrant and Ansible (2.0.0.2). It worked with Ansible < 2.

`dest_host` and `src_host` are both 127.0.0.1 but `ansible_ssh_port` is 2222. Now https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/synchronize.py#L175 fails and `between_multiple_hosts` is `False`. This leads to an error.

The fix from @nomaster looks good. Could you send a PR? Thanks.

**Comment by nomaster** [comment]
I'm sorry @simono, but my code is merely a workaround. It figures that an SSH port other than 22 leads to a remote system, which is not a universal truth. At this time, I cannot think of a better solution.

Does anyone have an idea?

**Comment by abadger** [comment]
This should be fixed in devel and the stable-2.0 branch (will be ansible-2.0.1) via https://github.com/ansible/ansible/pull/14140

**Comment by simono** [comment]
Thanks @abadger for the fast fix. Highly appreciated!
